### PR TITLE
Update GitLab Runner Instance Type

### PR DESCRIPTION
**Pull request by jonryser**
PR Checklist:
[X] Describe and explain your intentions for this change
GitLab runner instance type updates are working in Staging. Moving them to Prod.
Note that the DBInstanceClass was `db.t3.large`, but is now `db.r6g.large`. This change must not have made it to Prod previously. See previous PRs #51 and #52 

[X] Setup pre-commit and run the validators (info in README.md)
    To validate files run: `pre-commit run --all-files`

### CODE REVIEW

File: config/prod/iatlas-db.yaml [hunk 1]
```diff
@@ -10,5 +10,5 @@ parameters:
   KmsKey: !stack_output_external iatlas-kms-prod::KmsKey
   DbSecretPrefix: 'prod_'
   DbCredsName: 'iatlas_db_creds_encrypted_prod'
-  DBInstanceClass: 'db.t3.large'
+  DBInstanceClass: 'db.r6g.large'
```
Comment (zaro0508) [inline config/prod/iatlas-db.yaml:13]: Why not R7g or T4? https://aws.amazon.com/rds/instance-types/

File: config/config.yaml [hunk 2]
```diff
@@ -9,10 +9,10 @@ gitlab_container_pass: {{ environment_variable.GITLAB_CONTAINER_PASS | default('
 gitlab_container_user: {{ environment_variable.GITLAB_CONTAINER_USER | default('_WILL_NOT_WORK_') }}
 # DO NOT store the token in git.
 gitlab_reg_token: {{ environment_variable.GITLAB_REG_TOKEN | default('_NEVER_GONNA_GET_IT_') }}
-gitlab_runner_instance_type_prod: 'r5a.2xlarge'
+gitlab_runner_instance_type_prod: 'r5a.4xlarge'
```
Comment (zaro0508) [inline config/config.yaml:12]: why not M7g or R7g?  https://aws.amazon.com/ec2/instance-types/

File: templates/gitlab-manager-runner-roles.yaml [hunk 3]
```diff
@@ -33,12 +33,11 @@ Parameters:
...
       - 't3a.2xlarge'
-      - 'c5d.large'
-      - 'c5d.xlarge'
-      - 'c5d.2xlarge'
       - 'r5a.large'
       - 'r5a.xlarge'
       - 'r5a.2xlarge'
+      - 'r5a.4xlarge'
+      - 'r5a.8xlarge'
     Default: 't3a.micro'
```
Comment (zaro0508) [inline templates/gitlab-manager-runner-roles.yaml:40]: maybe update to the same instance type as you are going to use for staging and prod?

**Review by zaro0508 — COMMENTED** [review]
seems ok, but might be better for cost and performance to go with newer instance types.

File: config/config.yaml (hunk 2 above)
Comment (jonryser) [reply config/config.yaml:12]: We are needing memory optimized. The `r7g.4xlarge` is a little bit less expensive than the`r5a.4xlarge` (but only $.05/hr). Though, it is ARM based and makes me a little concerned about software compatibility.
I would still want to go with the r5a.

File: config/prod/iatlas-db.yaml (hunk 1 above)
Comment (jonryser) [reply config/prod/iatlas-db.yaml:13]: We are needing memory optimized. It looks like the `r7g` is a great option, has a bit more network bandwidth, but is also a little bit more expensive. Thoughts on why the `r7g` over the `r6g`?

File: templates/gitlab-manager-runner-roles.yaml (hunk 3 above)
Comment (jonryser) [reply templates/gitlab-manager-runner-roles.yaml:40]: Sure. We don't really need to list `r5a.8xlarge` in there.

File: config/prod/iatlas-db.yaml (hunk 1 above)
Comment (xschildw) [reply config/prod/iatlas-db.yaml:13]: Yeah, I noticed that the last models are not cheaper than the previous ones nearly across the board. Unless you need the extra perf, I think it's fine.

File: config/config.yaml (hunk 2 above)
Comment (xschildw) [reply config/config.yaml:12]: +1 on ARM, have the same concern.

File: templates/gitlab-manager-runner-roles.yaml (hunk 3 above)
Comment (jonryser) [reply templates/gitlab-manager-runner-roles.yaml:40]: `r5a.8xlarge` has been removed.

**Review by zaro0508 — APPROVED** [review]

**Review by xschildw — APPROVED** [review]

**Pull request merged**